nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -740,6 +753,64 @@ impl DataStore {
             })
     }
 
+    pub async fn nexus_inbound_icmp_view(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<ServiceIcmpConfig, Error> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+        use nexus_db_schema::schema::vpc_firewall_rule::dsl;
+
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        let rule = dsl::vpc_firewall_rule
+            .filter(dsl::time_deleted.is_null())
+            .filter(dsl::vpc_id.eq(*SERVICES_VPC_ID))
+            .filter(dsl::name.eq(NEXUS_ICMP_FW_RULE_NAME.to_string()))
+            .limit(1)
+            .select(VpcFirewallRule::as_select())
+            .get_result_async::<VpcFirewallRule>(&*conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(ServiceIcmpConfig {
+            enabled: rule.status.0 == VpcFirewallRuleStatus::Enabled,
+        })
+    }
+
+    pub async fn nexus_inbound_icmp_update(
+        &self,
+        opctx: &OpContext,
+        config: ServiceIcmpConfig,
+    ) -> Result<ServiceIcmpConfig, Error> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+        use nexus_db_schema::schema::vpc_firewall_rule::dsl;
+
+        let ServiceIcmpConfig { enabled } = config;
+
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        let status = nexus_db_model::VpcFirewallRuleStatus(if enabled {
+            VpcFirewallRuleStatus::Enabled
+        } else {
+            VpcFirewallRuleStatus::Disabled
+        });
+
+        let now = Utc::now();
+        let rule = diesel::update(dsl::vpc_firewall_rule)
+            .filter(dsl::time_deleted.is_null())
+            .filter(dsl::vpc_id.eq(*SERVICES_VPC_ID))
+            .filter(dsl::name.eq(NEXUS_ICMP_FW_RULE_NAME.to_string()))
```

Comment:
similarly i think we ought to be able to use an `&'static str` here:
```suggestion
            .filter(dsl::name.eq(NEXUS_ICMP_FW_RULE_NAME))
```